packages/blockchain-environment-v1/src/FabricEnvironmentConnection.ts

```diff
@@ -432,8 +432,17 @@ export class FabricEnvironmentConnection implements IFabricEnvironmentConnection
 
     private async setNodeContext(nodeName: string): Promise<void> {
         const node: FabricNode = this.getNode(nodeName);
-        const fabricWallet: IFabricWallet = await this.getWallet(nodeName);
-        await fabricWallet['setUserContext'](this.client, node.identity);
+        const fabricWallet: FabricWallet = await this.getWallet(nodeName);
+        const fabricIdentity: FabricIdentity = await fabricWallet.getIdentity(node.identity);
+        const identity: any = {
+            credentials: {
+                certificate: fabricIdentity.cert,
+                privateKey: fabricIdentity.private_key,
+            },
+            mspId: fabricIdentity.msp_id
+        };
+
+        await fabricWallet.getWallet()['getProviderRegistry']()['getProvider']('X.509')['setUserContext'](this.client, identity, node.identity);
```

Comment:
Anyway we can make this a bit nicer? :eyes:

Reply:
no it needs to be like that for now but will all change in my next pr, it being like that is just how it has to be for now